### PR TITLE
ci: #145 eslint setup

### LINKED ISSUE #145: Replace tslint usage with eslint.
Note that `tslint` is deprecated and should be replaced with `eslint`, I plan to do that in the future.

_Originally posted by @miquelbeltran in https://github.com/MindscapeHQ/raygun4node/pull/144#discussion_r1577255744_

### PR DESCRIPTION
**Pull request by miquelbeltran**
### Description :memo:

Removes deprecated `tslint` and setups `eslint`.

- Closes #145 

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**Updates**

- Removed `tslint` and config file.
- Install `eslint` with support for typescript.
- Setup `eslint` default config.
  - Added rules to support the current codestyle without changes.
- Setup `eslint` command
  - `npm run eslint` to run on `lib` and `test`.
  - Also the files in the `examples` folder are included.
- Setup GitHub Actions to run `eslint`.

Note: The current `eslint` config is set up in a way that no code changes are required. We can fix the code changes and remove the ignored rules in future PRs.

### Test plan :test_tube:

- No code changes done, only dev setup changes.
- CI jobs should pass, which now include running `eslint` step.

### Author to check :eyeglasses:
- [x] Project and all contained modules builds successfully
- [x] Self-/dev-tested 
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:
- [x] Project and all contained modules builds successfully
- [ ] Change has been dev-/reviewer-tested, where possible
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        rules: {
+            // TODO: Remove ignored rules and fix the code
+            "@typescript-eslint/no-explicit-any": "off",
+            "@typescript-eslint/no-this-alias": "off",
+            "@typescript-eslint/no-unused-vars": "off",
+            "@typescript-eslint/no-var-requires": "off",
+            "no-undef": "off",
+            "no-unreachable": "off",
+            "no-useless-catch": "off",
+        }
```
Comment (miquelbeltran) [inline eslint.config.mjs:19]: These are rules that the original `tslint` ignored and/or were not covered by the defaults. They have been added, so the code passes lint checks "as is".

For example, here I set: `"@typescript-eslint/no-explicit-any": "off"` and the original `tslint.json` had: `"no-unsafe-any": false,` 

My plan is to enable more severe rules as we move forward. The current configuration works as a baseline.

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// @ts-check
+
+import eslint from '@eslint/js';
+import tseslint from 'typescript-eslint';
+
+export default tseslint.config(
+    eslint.configs.recommended,
+    ...tseslint.configs.recommended,
```
Comment (miquelbeltran) [inline eslint.config.mjs:8]: Provides the basic recommended rules for JS and TS

File: eslint.config.mjs (hunk 1 above)
Comment (sumitramanga) [reply eslint.config.mjs:19]: Will this break the customers code?

**Review by sumitramanga — COMMENTED** [review]
Just one comment from me otherwise this looks good! It's great to see us getting rid of the deprecated tslint and replacing it. Well done @miquelbeltran 🌟

File: eslint.config.mjs (hunk 1 above)
Comment (miquelbeltran) [reply eslint.config.mjs:19]: Nope, on the contrary, these rules only affect the code in the plugin, which remains the same at the moment.

**Comment by miquelbeltran** [comment]
Thank you @sumitramanga! I answered the question.

**Review by sumitramanga — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**